### PR TITLE
Dunder methods to improve objects

### PR DESCRIPTION
**Pull request by felipevolpone**
How to use dunder methods to add behavior to objects. I add an example of how implement a french deck that can be used as a list.

### CODE REVIEW

File: deck_as_list.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    def __getitem__(self, position):
+        return self._cards[position]
+
+card_a = Card('A', 'spades')
+print card_a
+
+deck = Deck()
+len(deck)
+
+print deck[0]
```
Comment (brennerm) [inline deck_as_list.py]: change to print()

File: deck_as_list.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+print deck[-1]
```
Comment (brennerm) [inline deck_as_list.py]: change to print()

File: deck_as_list.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+for card in deck:
+    print card
```
Comment (brennerm) [inline deck_as_list.py]: change to print()

File: deck_as_list.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+                                        for rank in self.ranks]
+
+    def __len__(self):
+        return len(self._cards)
+
+    def __getitem__(self, position):
+        return self._cards[position]
+
+card_a = Card('A', 'spades')
+print card_a
```
Comment (brennerm) [inline deck_as_list.py]: change to print()

**Comment by felipevolpone** [comment]
@brennerm It's done!

**Comment by brennerm** [comment]
Thanks!

**Pull request merged**